### PR TITLE
Stability fixes

### PR DESCRIPTION
**Pull request by GregoryMaks**
Fixed a few issues:
1. Issue with GestureRecognizer which prevented gesture recognizing when elements were added to the header
2. Added `Device.BeginInvokeOnMainThread` for code to be more secure in multithreaded environment

### CODE REVIEW

File: src/XFExpandableListView/Controls/ExpandableListView.cs [hunk 1]
```diff
@@ -206,16 +209,19 @@ public virtual async Task Expand(int position)
             /* Hard operations must not affect the UI Thread */
             await Task.Run(() =>
             {
-                foreach (var item in expandableGroup)
+                Device.BeginInvokeOnMainThread(() =>
```
Comment (ChasakisD) [inline src/XFExpandableListView/Controls/ExpandableListView.cs]: I think there is no point on creating a new thread which will executes on the main thread. Just remove the task.run

File: src/XFExpandableListView/Controls/ExpandableListView.cs [hunk 2]
```diff
@@ -229,16 +235,19 @@ public virtual async Task Collapse(int position)
...
```
Comment (ChasakisD) [inline src/XFExpandableListView/Controls/ExpandableListView.cs]: Same here. Just remove the task.run

File: src/XFExpandableListView/Cells/ExpandableGroupCell.cs [hunk 3]
```diff
@@ -64,58 +64,59 @@ protected override void OnChildAdded(Element child)
...
-                        {
-                            await ExpandableController.ToggleGroup(GroupController);
-                        }
+                    #endregion
 
-                        #endregion
+                    #region [Execute Cell Command]
 
-                        #region [Execute Cell Command]
+                    /* Execute the cell command */
```
Comment (ChasakisD) [inline src/XFExpandableListView/Cells/ExpandableGroupCell.cs]: There were some updates here. Try to pull and commit the changes again :)

**Review by ChasakisD — COMMENTED** [review]
Pointed out some little things :)

File: src/XFExpandableListView/Cells/ExpandableGroupCell.cs [hunk 4]
```diff
@@ -64,58 +64,59 @@ protected override void OnChildAdded(Element child)
...
+                    if (CellCommand == null) return;
```
Comment (ChasakisD) [inline src/XFExpandableListView/Cells/ExpandableGroupCell.cs]: Made some changes here in the meantime. Try pull and commit the changes again :)

File: src/XFExpandableListView/Controls/ExpandableListView.cs (hunk 1 above)
Comment (ChasakisD) [inline src/XFExpandableListView/Controls/ExpandableListView.cs]: No need to create a new thread that will be executed on the main thread. Just get it out from the Task.Run

File: src/XFExpandableListView/Controls/ExpandableListView.cs (hunk 2 above)
Comment (ChasakisD) [inline src/XFExpandableListView/Controls/ExpandableListView.cs]: Same here, just get it out from the Task.Run

File: src/XFExpandableListView/Cells/ExpandableGroupCell.cs (hunk 4 above)
Comment (GregoryMaks) [reply src/XFExpandableListView/Cells/ExpandableGroupCell.cs]: Merge was a bit incorrect, fixed that

File: src/XFExpandableListView/Controls/ExpandableListView.cs (hunk 1 above)
Comment (GregoryMaks) [reply src/XFExpandableListView/Controls/ExpandableListView.cs]: Removed

File: src/XFExpandableListView/Controls/ExpandableListView.cs (hunk 2 above)
Comment (GregoryMaks) [reply src/XFExpandableListView/Controls/ExpandableListView.cs]: Removed

**Comment by AlexSchuetz** [comment]
I can confirm, that the Exception is gone. Unfortunately I couldn't get the toggle-icon to work.
When do you think this fix will be available via nuget?

**Comment by ChasakisD** [comment]
@AlexSchuetz I will test it and push it to nuget asap. What do you mean that the toggle-icon is not working? The group expands but the icon is not rotating?

**Comment by AlexSchuetz** [comment]
Yes exactly. I got a Exception on the sample that "This" is unknown. I just changed the following line:

`<cells:ExpandableGroupCell CellCommand="{Binding Source={x:Reference This}, Path=BindingContext.ToggleCommand}" CellCommandParameter="{x:Reference IconLabel}">`

to this:

`<cells:ExpandableGroupCell CellCommand="{Binding ToggleCommand}" CellCommandParameter="{x:Reference IconLabel}">`

But the ToggleCommand isn't called, so my binding is wrong.

**Comment by AlexSchuetz** [comment]
By the way: I also tested successfully on iOS. I will test UWP as soon as the nuget is available and make a PR with an iOS and UWP sample as soon as I have a bit of time.

**Comment by GregoryMaks** [comment]
Add ‘This’ to your page’s XAML. It’s not working because it is not referencing anything.

**Comment by ChasakisD** [comment]
@AlexSchuetz Actually what I am doing in the binding there, is that I am referencing the same instance of the page. So x:Name="This" is needed to the current page. Just Binding ToggleCommand will probably not work, because it will be looking into the item of the itemsource, because viewcell got its bindingcontext from the item.

**Comment by AlexSchuetz** [comment]
If using defining everything in a view and not a page using 'This' causes an Exception:
`Xamarin.Forms.Xaml.XamlParseException`

After a bit of research I fixed my sample using the following lines:

    <ContentView xmlns="http://xamarin.com/schemas/2014/forms" 
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             ...
             x:Name="Container"
             x:Class="XFExpandableListViewSample.Views.ExpandableListView">
    ...
    <cells:ExpandableGroupCell CellCommand="{Binding Source={x:Reference Name=Container}, Path=BindingContext.ToggleCommand}" CellCommandParameter="{x:Reference IconLabel}">

**Comment by ChasakisD** [comment]
Glad to hear it! I will do my best to find some free time and push it to nuget today!

PR Accepted and Merged!

**Comment by AlexSchuetz** [comment]
I really appreciate your efford and work. Thank you.

**Comment by ChasakisD** [comment]
@AlexSchuetz 1.0.1 version published on nuget. It will be listed soon. :)

**Comment by AlexSchuetz** [comment]
Got it. Works link a charm. :)

**Pull request merged**